docs/expression_syntax.md

```diff
@@ -0,0 +1,195 @@
+**Is your feature request related to a problem? Please describe.**
+As part of a larger feature (#522) to support complex condition statements in data prepper there is a need to define a syntax for conditional statements. A conditional statement is a String that is evaluated at runtime and may reference fields within a record.
+
+**Describe the solution you'd like**
+_Terms used throught this document are defined in the Definitions section_
+
+## Supported Operators
```

Comment:
I think we can make this documentation quite a bit more straighforward for pipeline authors.